diff --git a/configure.ac b/configure.ac
index 259dc4f..2fd9a8b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -74,18 +74,7 @@ AM_PATH_PYTHON
 AC_ARG_WITH([pygi_overrides_dir],
             AC_HELP_STRING([--with-pygi-overrides-dir], [Path to pygobject overrides directory]))
 
-AC_MSG_CHECKING(for pygobject overrides directory)
-if test "x$with_pygi_overrides_dir" = "x" ; then
-        overrides_dir="`$PYTHON -c 'import gi; print(gi._overridesdir)' 2>/dev/null`"
-    # fallback if the previous failed
-	if test "x$overrides_dir" = "x" ; then
-	    overrides_dir="${pyexecdir}/gi/overrides"
-	fi
-else
-	overrides_dir="$with_pygi_overrides_dir"
-fi
-
-PYGI_OVERRIDES_DIR="$overrides_dir"
+PYGI_OVERRIDES_DIR="$with_pygi_overrides_dir"
 AC_SUBST(PYGI_OVERRIDES_DIR)
 AC_MSG_RESULT($PYGI_OVERRIDES_DIR)
 
