From bd5d33578122d4aa0fa5f13dccf17e1aeae8049e Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval R <msandova@gnome.org>
Date: Fri, 6 Oct 2023 17:27:54 +0200
Subject: [PATCH 1/2] notification: Move disconnect to dispose

---
 libnotify/notification.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libnotify/notification.c b/libnotify/notification.c
index 697dc81..3fd261a 100644
--- a/libnotify/notification.c
+++ b/libnotify/notification.c
@@ -397,12 +397,18 @@ notify_notification_dispose (GObject *object)
 {
         NotifyNotification        *obj = NOTIFY_NOTIFICATION (object);
         NotifyNotificationPrivate *priv = obj->priv;
+        GDBusProxy                *proxy;
 
         if (priv->portal_timeout_id) {
                 g_source_remove (priv->portal_timeout_id);
                 priv->portal_timeout_id = 0;
         }
 
+        proxy = _notify_get_proxy (NULL);
+        if (proxy != NULL && priv->proxy_signal_handler != 0) {
+                g_signal_handler_disconnect (proxy, priv->proxy_signal_handler);
+        }
+
         g_clear_object (&priv->icon_pixbuf);
 
         G_OBJECT_CLASS (parent_class)->dispose (object);
@@ -413,7 +419,6 @@ notify_notification_finalize (GObject *object)
 {
         NotifyNotification        *obj = NOTIFY_NOTIFICATION (object);
         NotifyNotificationPrivate *priv = obj->priv;
-        GDBusProxy                *proxy;
 
         _notify_cache_remove_notification (obj);
 
@@ -434,11 +439,6 @@ notify_notification_finalize (GObject *object)
         if (priv->hints != NULL)
                 g_hash_table_destroy (priv->hints);
 
-        proxy = _notify_get_proxy (NULL);
-        if (proxy != NULL && priv->proxy_signal_handler != 0) {
-                g_signal_handler_disconnect (proxy, priv->proxy_signal_handler);
-        }
-
         g_free (obj->priv);
 
         G_OBJECT_CLASS (parent_class)->finalize (object);
-- 
2.41.0

